test/pylib/scylla_cluster.py

```diff
@@ -295,19 +296,16 @@ async def cql_is_up(self) -> bool:
             caslog.setLevel(oldlevel)
         # Any other exception may indicate a problem, and is passed to the caller.
 
-    async def rest_api_is_up(self) -> bool:
-        """Test that the Scylla REST API is serving. Can be used as a
-        checker function at start up."""
+    async def get_host_id(self, api: ScyllaRESTAPIClient) -> bool:
+        """Try to get the host id (also tests Scylla REST API is serving)"""
         try:
-            async with aiohttp.ClientSession() as session:
-                url = f"http://{self.hostname}:10000/"
-                async with session.get(url):
-                    return True
-        except aiohttp.ClientConnectionError:
+            self.host_id = await api.get_host_id(self.hostname)
+            return True
+        except (aiohttp.ClientConnectorError, RuntimeError):
```

Comment:
I can use a custom exception. Is this OK?

Reply:
I don't know. I need to understand how Scylla and this code will behave when there's a problem and `/storage_service/hostid/local` throws an exception. We cannot come up with a correct solution before we understand the code.

Therefore I'd like this questions answered:
What will happen if Scylla has a problem throws an exception from the /storage_service/hostid/local endpoint? How will the code behave right now?

Reply:
We discussed this on Slack.

So it turns out that while Scylla is initializing, trying to fetch `/storage_service/hostid/local` might return `404` - the endpoint wasn't created yet.

We want to:
- retry if the error code is `404`
- fail if the error code is `500`

since `RuntimeError` loses this information, we decided that `RESTClient` should throw a better error type from `_fetch` - which doesn't lose this information.

Reply:
Done in https://github.com/scylladb/scylladb/commit/8c39b3b7dc7a3d956837eb340c75e00a5b4589a9